### PR TITLE
[android] turn on USE_VULKAN for android builds by default

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51291 [android] turn on USE_VULKAN for android builds by default**

Turning on USE_VULKAN for android builds
Remove standalone android vulkan build

Testing all ci jobs (for master): https://github.com/pytorch/pytorch/pull/51292

Differential Revision: [D26141891](https://our.internmc.facebook.com/intern/diff/D26141891)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -219,7 +219,9 @@ option(USE_TENSORRT "Using Nvidia TensorRT library" OFF)
 cmake_dependent_option(
     USE_VALGRIND "Use Valgrind. Only available on Linux." ON
     "LINUX" OFF)
-option(USE_VULKAN "Use Vulkan GPU backend" OFF)
+cmake_dependent_option(
+    USE_VULKAN "Use Vulkan GPU backend" ON
+    "ANDROID" OFF)
```
Comment (AshkanAliabadi) [inline CMakeLists.txt:224]: Is USE_VULKAN dependent on ANDROID being OFF or ON?  Will this be OK for desktop builds?

Comment (IvanKobzarev) [reply CMakeLists.txt:224]: https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html
Here the argument positions are non trivial: the default value is the latest, ON if ANDROID

Comment (AshkanAliabadi) [reply CMakeLists.txt:224]: Got your reply in an email, so is my understanding correct that this statement means: give the users the option to set USE_VULKAN to ON or OFF if ANDROID is OFF, right?

Comment (IvanKobzarev) [reply CMakeLists.txt]: Yes, I deleted my reply to verify if overrides with -DUSE_VULKAN work correct.

And in that case it does not. It is ON if android and OFF else.

I will try to find the way how to make override work

Comment (IvanKobzarev) [reply CMakeLists.txt]: @AshkanAliabadi , updated to construction
```
if(NOT DEFINED USE_VULKAN)
  cmake_dependent_option(
      USE_VULKAN "Use Vulkan GPU backend" ON
      "ANDROID" OFF)
endif()
```
which works fine with -DUSE_VULKAN=

File: CMakeLists.txt [hunk 2]
```diff
@@ -219,7 +219,13 @@ option(USE_TENSORRT "Using Nvidia TensorRT library" OFF)
...
+
+if(NOT DEFINED USE_VULKAN)
+  cmake_dependent_option(
+      USE_VULKAN "Use Vulkan GPU backend" ON
+      "ANDROID" OFF)
+endif()
+
 option(USE_VULKAN_FP16_INFERENCE "Vulkan - Use fp16 inference even on fp32 tensors" OFF)
 option(USE_VULKAN_RELAXED_PRECISION "Vulkan - Use relaxed precision math in the kernels (mediump)" OFF)
 option(USE_VULKAN_SHADERC_RUNTIME "Vulkan - Use runtime shader compilation (needs libshaderc)" OFF)
```
Comment (AshkanAliabadi) [inline CMakeLists.txt:231]: I think it will also makes sense to make these series of flags dependent on USE_VULKAN being ON.  What do you think

**Review by AshkanAliabadi — APPROVED** [review]
Thanks Ivan!

File: CMakeLists.txt (hunk 2 above)
Comment (IvanKobzarev) [reply CMakeLists.txt:231]: I think for the first iteration we can keep it like we have - USE_VULKAN works as a master switch anyway.

**Pull request closed without merging**